README.md

```diff
@@ -23,396 +23,345 @@ Stable release versions are manually released from the main branch.
-];
-const stakingDuration: number = 144;
-const stakingAmount: number = 1000;
-const unbondingTime: number = minUnbondingTime;
-
-// 3. Define the parameters for the staking transaction that will contain the
-//    staking contract:
-//    - `inputUTXOs: UTXO[]`: The list of UTXOs that will be used as an input
-//       to fund the staking transaction.
-//    - `feeRate: number`: The fee per tx byte in satoshis.
-//    - `changeAddress: string`: BTC wallet change address, Taproot or Native
-//       Segwit.
-//    - `network: network to work with, either networks.testnet
-//       for BTC Testnet and BTC Signet, or networks.bitcoin for BTC Mainnet.
-
-// Each object in the inputUTXOs array represents a single UTXO with the following properties:
-// - txid: transaction ID, string
-// - vout: output index, number
-// - value: value of the UTXO, in satoshis, number
-// - scriptPubKey: script which provides the conditions that must be fulfilled for this UTXO to be spent, string
-const inputUTXOs = [
-  {
-    txid: "e472d65b0c9c1bac9ffe53708007e57ab830f1bf09af4bfbd17e780b641258fc",
-    vout: 2,
-    value: 9265692,
-    scriptPubKey: "0014505049839bc32f869590adc5650c584e17c917fc",
-  },
-];
-const feeRate: number = 18;
-const changeAddress: string = btcWallet.address;
-const network = networks.testnet;
-```
+We will be using the full list of staking parameters in this library, for detailed
+explanation of each fields, refer to this [documentation](https://github.com/babylonlabs-io/babylon/blob/main/docs/register-bitcoin-stake.md#32-babylon-chain-btc-staking-parameters)
 
-### Fee Calculation
-The fee calculation in the @babylonlabs-io/btc-staking-ts library is based on an estimated size 
-of the transaction in virtual bytes (vB). This estimation helps in calculating 
-the appropriate fee to include in the transaction to ensure it is processed by 
-the Bitcoin network efficiently.
+#### Staker information
 
-The fee estimation formula used is:
-```
-numInputs * 180 + numOutputs * 34 + 10 + numInputs + 40
-```
+The library require Staker's BTC as well as their Babylon Genesis Account 
+information to be supplied in order to create the relevant transactions.
 
-This accounts for:
-- `180 vB` per input
-- `34 vB` per output
-- `10 vB` fixed buffer
-- `numInputs` additional factor
-- `40 vB` buffer for the op_return output
+1. **Staker's Bitcoin and Babylon Details**
+```ts
+const stakerInfo = {
+  address: "{{ Staker BTC address }}",
+  publicKeyNoCoordHex: "{{ Staker BTC public key without coordinates in hex }}",
+}
 
-### Create the Staking Contract
+const babylonAddress = "{{ Babylon Genesis Chain bech32 Address }}"
+```
 
-After defining its parameters,
-the staking contract can be created.
-First, create an instance of the `StakingScriptData` class
-and construct the Bitcoin scipts associated with Bitcoin staking using it.
+as well the staking inputs selected/created by the staker at the time of creating
+the delegation:
```

Comment:
What's a delegation? We haven't defined it. We can refer it as Bitcoin Staking transaction.